### DISCUSSION TITLE
The v0.19.0 update broke my samba backups via tailscale

### DISCUSSION THREAD

**Discussion by ovizii** [hassio-addons/app-tailscale#365] [Q&A] [answered]
I had a look at the changelog and am unsure what could be the reason. 
I have 2 HA instances which both create backups with https://github.com/thomasmauerer/hassio-addons/tree/master/samba-backup using SMB over tailscale and both broke as soon as I did this update.

The samba logs simply say:

```
[24-05-21 15:24:38] WARNING: do_connect: Connection to 192.168.192.110 failed (Error NT_STATUS_IO_TIMEOUT)
[24-05-21 15:24:38] FATAL: Cannot access share. Unknown reason.
```

If I go to my HA instance's terminal via Advanced SSH & Web Terminal I can ping 192.168.192.110

Any ideas?

**Comment by lmagyar** [comment]
I suggest to add to Tailscale config: `stateful_filtering: false`

TS v1.66.3 by default allow outgoing connections only from the tailscale0 interface, and that add-on doesn't run on the host network, so my guess is some routing is going on there.

Note: the next TS version, v1.66.4 will make it default off, so in the next add-on version you can remove this config option - if this is the problem. https://github.com/tailscale/tailscale/releases/tag/v1.66.4

**Reply by ovizii** [reply]
Thanks for having a look.

I'm not sure how to add a parameter to a config. Where do I find the config? 

I have tried enabling, Stateful packet filtering in the GUI, saving, restarting, disabling, saving restarting and the problem persists.

This is my current config:

![image](https://github.com/hassio-addons/addon-tailscale/assets/837902/fbae6b94-dc66-49a8-a4f3-170328b5afab)

**Comment by ovizii** [comment, marked as answer]
Oh, no, I have found the way to edit the config as yaml. This is mine:

```
accept_routes: true
advertise_exit_node: true
snat_subnet_routes: false
userspace_networking: false
log_level: warning
stateful_filtering: false
```

so it seems the problem is something else :-(

**Reply by lmagyar** [reply]
Yeah, the add-on config GUI is broken, you can't know whether it is turned off, or just doesn't configured. You have to enable something, save, disable, save, restart. I always use the yaml + restart. :)

Pffff, in theory, with this config it should work... In your SSH terminal enter: ``docker exec -it `docker ps -q -f name=samba-backup` /bin/bash``, now you are in the samba add-on's cli, and you can test the network config, ping, traceroute, etc. Enter `exit` to get back to the SSH add-on's cli. From now on it seems to me some network misconfig and/or TS bug. Hard to say anything more. :/

**Reply by ovizii** [reply]
Mysteriously, it started working after a few more saves and restarts. I'm still using this config. No changes to samba backup.

@lmagyar thanks for all the help, I can only assume, that `stateful_filtering: false` was the fix indeed.

**Comment by ovizii** [comment]
Thanks for the pointers, I tried but samba backup add-on will shut down if it cannot reach the smb server. I tried entering a different container but couldn't find any which had at least ping inside. Will keep checking and update this thread if I figure anything out.

**Comment by ovizii** [comment]
Saw some log lines which make me wonder whether I shouldn't be seeing any tailscale0 adapter in the logs?

```
2024/05/21 21:02:26 monitor: [unexpected] network state changed, but stringification didn't: interfaces.State{defaultRoute=enp0s18 ifs={docker0:[172.30.232.1/23 llu6] enp0s18:[192.168.96.2/28 llu6] hassio:[172.30.32.1/23 llu6]} v4=true v6=false}
2024/05/21 21:02:26 monitor: [unexpected] old: {"InterfaceIPs":{"docker0":
```

**Comment by lmagyar** [comment]
Maybe this is related to #362, where TS misteriously stops working, then starts working, then stops... Seems to be some TS issue. Or not.

**Reply by ovizii** [reply]
That sounds exactly like what I experienced :-/

**Comment by ovizii** [comment]
Btw. the v0.19.1 update fixed everything. I updated the add-on, restarted it and voilà, my backups are working again. No configuration changes.
